### PR TITLE
fix for #380 PLUS add tempdir usage to fetch

### LINKED ISSUE #380: landsat c1 cloudmask needs inversion and assignment of nodata value

### PR DESCRIPTION
**Pull request by ircwaves**
Main issue was that nodata value wasn't set on `'cloudmask'`, but in addition to that, inverting the mask to be a `good_data=1`, `bad_data=0` simplifies usage.  Also, observed that the cloudmask seemed to miss hazy edges of clouds, so added an erosion step to label more data as `bad` and avoid the inclusion of polluted pixels.  fixes #380

### CODE REVIEW

File: gips/data/landsat/landsat.py [hunk 1]
```diff
@@ -329,17 +329,32 @@ def fetch(cls, asset, tile, date):
...
+            url = api.download(
+                result['dataset'], 'EE', sceneIDs, 'STANDARD', api_key
+            )['data'][0]
+            with utils.make_temp_dir(prefix='dwnld', dir=stage_dir) as dldir:
+                download(url, dldir)
+                granules = os.listdir(dldir)
+                if len(granules) != 1:
+                    raise Exception(
+                        'Download appears to have not produced a file: {}'
+                        .format(str(granules))
```
Comment (ags-tolson) [inline gips/data/landsat/landsat.py]: It looks like when `len(granules) > 1`, the exception may have a confusing message.  Is it worth it to reword it?  Maybe:  `"Expected 1 thing, but got {} things: {things}".format(len(things), things)`

Also I believe `"{}".format(some_list)` does the `str()` part for you if you do decide to change this bit.

File: gips/data/landsat/landsat.py [hunk 2]
```diff
@@ -1001,20 +1016,43 @@ def process(self, products=None, overwrite=False, **kwargs):
...
+
+                        # We already have an implicit dependency on scipy (from
+                        # Py6S), but life might be more simple if we just move
+                        # the whole cloudmask extraction into gippy.
+                        from scipy import ndimage
+                        np_cloudmask_erroded = ndimage.binary_erosion(
+                            np_cloudmask,
+                            structure=numpy.ones((10, 10), dtype=numpy.uint8),
+                        )
+                        #
```
Comment (ags-tolson) [inline gips/data/landsat/landsat.py]: optional:  remove empty comment

File: gips/data/landsat/landsat.py [hunk 3]
```diff
@@ -1001,20 +1016,43 @@ def process(self, products=None, overwrite=False, **kwargs):
...
+                        # 
+
+                        imgout = gippy.GeoImage(fname, img, gippy.GDT_Byte, 1)
                         verbose_out("writing " + fname, 2)
-                        imgout.SetBandName('Cloud Mask', 1)
-                        imgout[0].Write(np_cloudmask)
+                        imgout.SetBandName(
+                            self._products[val[0]]['bands'][0], 1
+                        )
+                        imgout.SetMeta('GIPS_LANDSAT_VERSION', self.version)
```
Comment (ags-tolson) [inline gips/data/landsat/landsat.py]: Optional:  Maybe not now, but it seems like this should be set for all products

File: gips/data/landsat/landsat.py [hunk 4]
```diff
@@ -1001,20 +1016,43 @@ def process(self, products=None, overwrite=False, **kwargs):
...
+                        np_cloudmask = numpy.logical_not(
+                            get_bit(npqa, 4) &
+                            get_bit(npqa, 6) |
+                            get_bit(npqa, 8)
+                        )
+
+                        # We already have an implicit dependency on scipy (from
+                        # Py6S), but life might be more simple if we just move
+                        # the whole cloudmask extraction into gippy.
+                        from scipy import ndimage
```
Comment (ags-tolson) [inline gips/data/landsat/landsat.py]: optional:  Is it worth it to add scipy to gips dependencies so it's installed explicitly?  Also should this import go at the top of the file?

File: gips/data/landsat/landsat.py (hunk 1 above)
Comment (ircwaves) [reply gips/data/landsat/landsat.py]: A few lines back the code ensures that we only got one URL from the query_service, so provided `download(one_url)` implies we get  1 file, then the only case remaining is the no-file case.  Does that make sense.  I guess I will change it to `if not granules:`.  Seem safer?

File: gips/data/landsat/landsat.py (hunk 3 above)
Comment (ircwaves) [reply gips/data/landsat/landsat.py]: Agreed...it was a "while I'm here" fix.  We need a metadata standardization effort.

File: gips/data/landsat/landsat.py (hunk 1 above)
Comment (ircwaves) [reply gips/data/landsat/landsat.py]: err...I'll go `len(granules) == 0` for clarity.

**Pull request merged**